src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -117,7 +146,25 @@ internal RedisCache(IOptions<RedisCacheOptions> optionsAccessor, ILogger logger)
 
         token.ThrowIfCancellationRequested();
 
-        return await GetAndRefreshAsync(key, getData: true, token: token).ConfigureAwait(false);
+        var expandedKey = _expandKey(key);
```

Comment:
Note I've already improved the key handling code in an unmerged PR; I suggest I get that one finished ASAP in January, and you rebase from that one; either way, we're trampling all over eachother here - I guess if you're still around you win :)

Reply:
@mgravell Did that change go in?  Is this PR unblocked?

Reply:
Something else that might be necessary is to put back the support for older REDIS instances that I removed as I assumed that since even Azure only supports 6.0 it was not necessary anymore. But @adityamandaleeka has a reason to keep it.

Reply:
@sebastienros Do you remember what my reason was? 😆

Reply:
I believe you said @NickCraver had given you some good reasons. But it's starting to read like a comedy ;)

Reply:
"Most users are still on v4" would be my info contribution here :)